src/App/App.js

```diff
@@ -0,0 +1,202 @@
+    this.setState({ error: '' })
+  }
+
+  signOut = () => {
+    this.setState({ currentUser: null })
+  }
+
+  get filterMoviesBySelection() {
+    const filterByGenre = this.state.movies.filter(movie => {
+      if (this.state.dropdownValue !== 'all' && movie.genres) {
+        return movie.genres.includes(this.state.dropdownValue)
+      } else {
+        return this.state.movies
+      }
+    })
+
+    const filterByRating = filterByGenre.filter(movie => {
+      if (this.state.ratingValue !== 'any') {
+        const roundedRating = parseInt(movie.average_rating)
+        return roundedRating === parseInt(this.state.ratingValue)
+      } else {
+        return filterByGenre
+      }
+    })
+
+    return filterByRating.filter(movie => {
+      return movie.title.toLowerCase().includes(this.state.input)
+    })
+  }
+
+  getUserInput = (inputValue) => {
+    this.setState({input: inputValue})
+  }
+
+  handleDropdownValue = (event) => {
+    this.setState({ dropdownValue: event.target.name})
+  }
+
+  handleRatingValue = (event) => {
+    this.setState({ ratingValue: event.target.value })
+  }
+
+  clearFilters = () => {
+    this.setState({ input: '', dropdownValue: 'all', ratingValue: 'any' })
+  }
+
+  render() {
+    return(
+      <main>
+        <header>
+          <h1 className='page-title'>underrated</h1>
+          <nav>
+            <NavLink exact to="/" className='nav-labels'>
+              all movies
+            </NavLink>
+            <p className='nav-labels'>watchlist</p>
+            <DropdownButton 
+              className='nav-labels' 
+              title='account'>
+                {!this.state.currentUser &&
+                  <Login 
+                    validateLogin={this.validateLogin} 
+                    error={this.state.error} 
+                    clearError={this.clearError}
+                  />
+                }
+                {this.state.currentUser &&
+                  <section>
+                    <p>Hello, {this.state.currentUser.name}!</p>
+                    <button onClick={this.signOut}>Sign Out</button>
+                  </section>     
+                }
+            </DropdownButton>
+          </nav>
+        </header>
+        <Route 
+          exact 
+          path="/" 
+          render={() => {
+            return (
+              <section>
+                <Banner />
+                <Search 
+                  getUserInput={this.getUserInput} 
+                />
+                <section className="genre-filter">
+                  <p className="dropdown-text">showing <span className="active-text">{this.state.dropdownValue}</span> movies</p>
+                  <DropdownButton
+                    title=''
+                    className='genre-dropdown'>
+                    <button onClick={() => this.setState({ dropdownValue: 'all' })}>all</button>
+                    <button name="Action" onClick={(event) => this.handleDropdownValue(event)}>action</button>
```

Comment:
You should be able to write these like:
`<button name="Action" onClick={this.handleDropdownValue}>action</button>`

The event should come through when the `onClick` is called.